docs/geonetwork-synchronization.md

```diff
@@ -0,0 +1,61 @@
+# Synchronization between GeoNetwork and the LDAP
+
+GeoNetwork requires to create some objects in its database to function
+correctly, this mainly to ensure referential integrity. As a result, a
+[process](https://github.com/georchestra/datadir/blob/master/geonetwork/config/config-security-georchestra.xml#L108-L124)
+is configured to run at a regular pace, which ensures groups are created from
+some LDAP objects.
+
+By default, this process is configured to create groups from the organizations
+registered in the LDAP tree (the `ou=orgs` branch).
+
+Historically though, geOrchestra only had a `ou=groups` branch, and to
+distinguish between roles and groups, a prefix was used (`SV_` for roles, `EL_`
+for groups). Some organizations still want to stick with the previous
+  synchronization behaviour. This is still possible: by modifying the datadir
+  consequently:
+
+In the
+[config-security-georchestra.xml](https://github.com/georchestra/datadir/blob/master/geonetwork/config/config-security-georchestra.xml#L117)
+file, one have to change the filter to get the groups prefixed by `EL_`:
+
+
+```diff
+-    <entry key="ldapGroupSearchFilter" value="(objectClass=groupOfMembers)"/>
++    <entry key="ldapGroupSearchFilter" value="(&amp;(objectClass=groupOfMembers)(cn=EL_*))"/>
+
+```
+
+Also, to remove the `EL_` prefix, one have to adapt the default pattern, and
+map the label of the group onto the `description` attribute from the ldap:
+
+```diff
+-      <entry key="ldapGroupSearchPattern" value="(.*)"/>
+-      <entry key="ldapGroupLabelAttribute" value="o"/>
++     <entry key="ldapGroupSearchPattern" value="EL_(.*)"/>
++     <entry key="ldapGroupLabelAttribute" value="description"/>
+```
+
+Finally, we need to edit the
+[geonetwork.properties](https://github.com/georchestra/datadir/blob/master/geonetwork/geonetwork.properties#L60)
+to tell the synchronization process to use the `ou=roles` branch of the LDAP:
+
+```diff
+-ldap.groups.search.base=ou=orgs
++ldap.groups.search.base=ou=roles
+```
+
+# Automatic creation of groups
+
+GeoNetwork in geOrchestra is also configured to create automatically
+non-existing groups when the user connects onto the catalogue. With the
+previous configuration, mapping the geonetwork groups onto the `ou=roles` LDAP
+branch will create groups like `ADMINISTRATOR`, `REVIEWER` and so on at first
```

Comment:
Still dont get it - why would they be synced since `ldapGroupSearchPattern=EL_(.*)` ?

Reply:
because of this method:
https://github.com/georchestra/geonetwork/blob/ee5644710df42acc2900fb01c8b28f993b33afec/core/src/main/java/org/fao/geonet/kernel/security/ldap/LDAPUtils.java#L129-L175 Which does not take care of the pattern.

Note that the previous method is not called during the synchronization process, but at session creation (e.g. when the connected user first loads a geonetwork page). The `user.getPrivileges()` call line 134 returns a map indexed as  `group name => user profile for this group`, where we find back the elements from the LDAP branch. The main question is why this multimap is populated this way ?